.github/scripts/archive_eol.py

```diff
@@ -0,0 +1,71 @@
+import os
+import subprocess
+
+from github import Github
+from github.GithubException import RateLimitExceededException, UnknownObjectException
+
+
+def get_eol_refs(arch: str, remote: str) -> set:
+    cmd = (
+        f'flatpak remote-ls --user --arch="{arch}" --all --columns=ref,options {remote}'
+    )
+    ret = subprocess.run(cmd, shell=True, capture_output=True, encoding="utf-8")
+    eol_refs = set()
+
+    if ret.returncode != 0:
+        return
+
+    for line in ret.stdout.split("\n"):
+        if line:
+            split = line.split("\t")
+        if len(split) < 2:
+            continue
+        ref, options = split
+        ref = ref.split("/")[1]
+        if any(x in options for x in ("eol=", "eol-rebase=")) and not ref.endswith(
+            (".Debug", ".Sources", ".Locale")
+        ):
+            eol_refs.add(ref)
+
+    return eol_refs
+
+
+def main():
+
+    token = os.environ["GITHUB_TOKEN"]
+
+    g = Github(token)
+
+    # Exclude refs that have EOL notices but are still maintained
+    # in some branch
+    excludes = {
+        "org.kde.kate",
+        "org.mozilla.firefox.BaseApp",
+        "com.riverbankcomputing.PyQt.BaseApp",
+        "org.freedesktop.Sdk.Extension.mono6",
+        "org.freedesktop.Sdk.Extension.openjdk8",
+        "org.freedesktop.Sdk.Extension.rust-stable",
+        "org.freedesktop.Sdk.Extension.golang",
+        "org.freedesktop.LinuxAudio.BaseExtension",
+    }
```

Comment:
Wondering if we should restrict this to apps only. Some people will EOL extensions when they shouldn't be 

Even with apps we will still need some excludes.

With `runtimes` excluded, we will miss the countless plugin repos and theme extensions.